### PR TITLE
Fixing cspell spelling errors in spring packages

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.
Fixes: https://github.com/Azure/azure-sdk-for-java/issues/31612, https://github.com/Azure/azure-sdk-for-java/issues/31636, https://github.com/Azure/azure-sdk-for-java/issues/31638, https://github.com/Azure/azure-sdk-for-java/issues/31639, https://github.com/Azure/azure-sdk-for-java/issues/31641

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -201,13 +199,10 @@
     "sdk/search/azure-search-documents/**",
     "sdk/spring/spring-cloud-azure-starter-storage-blob/**",
     "sdk/spring/spring-cloud-azure-starter-servicebus/**",
-    "sdk/spring/spring-cloud-azure-starter-servicebus-jms/**",
     "sdk/spring/spring-cloud-azure-starter-storage-file-share/**",
     "sdk/spring/spring-cloud-azure-starter-storage-queue/**",
     "sdk/spring/spring-cloud-azure-starter-stream-eventhubs/**",
     "sdk/spring/spring-cloud-azure-starter-stream-servicebus/**",
```
Comment (saragluna) [inline .vscode/cspell.json:205]: what about these starters?

File: .vscode/cspell.json [hunk 2]
```diff
@@ -855,6 +853,168 @@
...
+        "ecwid",
+        "ehcache",
+        "epoll",
+        "fcgi",
+        "flywaydb",
+        "gcloud",
+        "groovysh",
+        "guice",
+        "Hikari",
+        "hikaricp",
```
Comment (saragluna) [inline .vscode/cspell.json:889]: Are all these in the scripts folder?

**Review by saragluna — APPROVED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cosmos/AzureCosmosAutoConfiguration.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public CosmosClient azureCosmosClient(CosmosClientBuilder builder) {
      */
     @Bean
     @ConditionalOnMissingBean
-    // TODO (xiada): spring data cosmos also defines a CosmosAsyncClient
+    // TODO: spring data cosmos also defines a CosmosAsyncClient
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cosmos/AzureCosmosAutoConfiguration.java:62]: I don't think this should be changed, it is just false alarm from the spell checking tool, the correct way is fixing it there

Comment (Netyyyy) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/cosmos/AzureCosmosAutoConfiguration.java:62]: Fixes in https://github.com/Azure/azure-sdk-for-java/pull/32844

**Pull request merged**